test/integration/cps/test_cps.py

```diff
@@ -117,3 +120,63 @@ def package_info(self):
     assert 'set(zlib_INCLUDE_DIRS_RELEASE "${zlib_PACKAGE_FOLDER_RELEASE}/include")' in cmake
     assert 'set(zlib_LIB_DIRS_RELEASE "${zlib_PACKAGE_FOLDER_RELEASE}/lib")'
     assert 'set(zlib_LIBS_RELEASE zlib)' in cmake
+
+
+def test_cps_merge():
+    folder = temp_folder()
+    cps_base = textwrap.dedent("""
+        {
+          "components" : {
+            "mypkg" : {
+              "includes" : [ "@prefix@/include" ],
+              "type" : "archive"
+            }
+          },
+          "cps_version" : "0.12.0",
+          "name" : "mypkg",
+          "version" : "1.0"
+        }
+        """)
+    cps_conf = textwrap.dedent("""
+        {
+          "components" : {
+            "mypkg" : {
+              "link_languages" : [ "cpp" ],
+              "location" : "@prefix@/lib/mypkg.lib"
+            }
+          },
+          "configuration" : "Release",
+          "name" : "mypkg"
+        }
+        """)
+    save_files(folder, {"mypkg.cps": cps_base,
+                        "mypkg@release.cps": cps_conf})
+    cps = CPS.load(os.path.join(folder, "mypkg.cps"))
+    json_cps = cps.serialize()
+    print(json.dumps(json_cps, indent=2))
+
+
+def test_extended_cpp_info():
+    c = TestClient()
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            def package_info(self):
+                self.cpp_info.libs = ["mylib"]
+                self.cpp_info.location = "my_custom_location"
+                self.cpp_info.type = "static-library"
+            """)
+    c.save({"conanfile.py": conanfile})
+    c.run("create .")
+
+    settings = "-s os=Windows -s compiler=msvc -s compiler.version=191 -s arch=x86_64"
+    c.run(f"install --requires=pkg/0.1 {settings} -g CPSDeps")
+    pkg = json.loads(c.load("build/cps/msvc-191-x86_64-release/pkg.cps"))
+    assert pkg["name"] == "pkg"
+    assert pkg["version"] == "0.1"
+    assert pkg["default_components"] == ["pkg"]
+    pkg_comp = pkg["components"]["pkg"]
+    assert pkg_comp["type"] == "archive"
+    assert pkg_comp["location"] == "my_custom_location"
```

Comment:
I am wondering what about `mylib` name, is it not saved in the `pkg.cps` file? If it is, I think that we could add another assertion.

Reply:
Not really, the ``location`` is what is really used, it should contain the ``mylib`` name as full filename. In reality, for CPS the linkname ``-lmylib`` is not existing it will be deduced back by tooling and buildsystem from the ``location``